app/controllers/projects_controller.rb

```diff
@@ -72,6 +86,20 @@ def set_project
     @project = Project.includes(:configs).find(params[:id])
   end
 
+  def set_project_metrics
+    @project_metrics = {}
+    ProjectMetrics.metric_names.each do |name|
+      @project_metrics[name] = ProjectMetrics.class_for name
+    end
+  end
+
```

Comment:
slightly more rubyish version:

```rb
  def set_project_metrics
    @project_metrics = ProjectMetrics.metric_names.inject({}) do |hash, name|
      hash[name] = ProjectMetrics.class_for name; hash
    end
  end
```

see http://stackoverflow.com/a/9434319/316729 for more alternatives